spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/jdbc/JndiDataSourceAutoConfigurationTests.java

```diff
@@ -80,7 +79,7 @@ void close() {
 	}
 
 	@Test
-	void dataSourceIsAvailableFromJndi() throws IllegalStateException, NamingException {
+	void dataSourceIsAvailableFromJndi() throws IllegalStateException {
```

Comment:
`IllegalStateException` is unchecked so it doesn't need to be declared either. It'd be nice to get rid of those too if you have the time.

Reply:
Okay, let me do it, I think there may be many similar problems in this project, I will check them one by one in the next two days, please do not merge too early

Reply:
Thanks very much. I'll mark this as a draft for now so that we know it's not quite ready to be merged.